src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/AddStatisticsCardView/AddStatisticsCardView.swift

```diff
@@ -0,0 +1,108 @@
+	
+	// swiftlint:disable:next function_parameter_count
+	func configure(
+		localStatisticsModel: AddLocalStatisticsModel,
+		availableCardsState: CreatedLocalStatisticsState,
+		onAddStateButtonTap: @escaping (SelectValueTableViewController) -> Void,
+		onAddDistrict: @escaping (SelectValueTableViewController) -> Void,
+		onDismissState: @escaping () -> Void,
+		onDismissDistrict: @escaping (Bool) -> Void,
+		onEditButtonTap: @escaping () -> Void,
+		onAccessibilityFocus: @escaping () -> Void
+	) {
+		self.model = localStatisticsModel
+
+		self.onAddStateButtonTap = onAddStateButtonTap
+		self.onAddDistrict = onAddDistrict
+		self.onDismissState = onDismissState
+		self.onDismissDistrict = onDismissDistrict
+		self.onEditButtonTap = onEditButtonTap
+		self.onAccessibilityFocus = onAccessibilityFocus
+	}
+
+	@IBAction func addLocalIncidencePressed(_ sender: Any) {
+		guard let model = self.model else {
+			Log.warning("AddStatistics model is nil", log: .localStatistics)
+			return
+		}
+		viewModel = AddStatisticsCardsViewModel(
+			localStatisticsModel: model,
+			presentSelectStateList: { selectedStateValueViewModel in
+				self.presentAddLocalStatistics(selectValueViewModel: selectedStateValueViewModel)
+			},
+			presentSelectDistrictList: { selectedDistrictValueViewModel in
+				self.presentAddLocalStatisticsDistrict(selectValueViewModel: selectedDistrictValueViewModel)
+				
+			}
+		)
+		viewModel?.presentStateSelection()
+	}
+	
+	@IBOutlet weak var addLocalIncidenceLabel: ENALabel!
+	// WE NEED LocalStatisticsModel INJECTION
+	// WE NEED COMPLETION HANDLERS FOR DISMISS AND SHOW
+
+	// MARK: - Private
+
+	private func presentAddLocalStatistics(selectValueViewModel: SelectValueViewModel) {
+		let selectValueViewController = SelectValueTableViewController(
+			selectValueViewModel,
+			closeOnSelection: false,
+			dismiss: { [weak self] in
+				self?.onDismissState?()
+			})
+		onAddStateButtonTap?(selectValueViewController)
+	}
+
+	private func presentAddLocalStatisticsDistrict(selectValueViewModel: SelectValueViewModel) {
+		let selectValueViewController = SelectValueTableViewController(
+			selectValueViewModel,
+			closeOnSelection: true,
+			dismiss: { [weak self] in
+				let dismissToRoot = self?.viewModel?.district == nil ? false : true
+				self?.onDismissDistrict?(dismissToRoot)
+			}
+		)
+		onAddDistrict?(selectValueViewController)
+	}
+
+	private var onAddStateButtonTap: ((SelectValueTableViewController) -> Void)?
+	private var onAddDistrict: ((SelectValueTableViewController) -> Void)?
+	private var onDismissState: (() -> Void)?
+	private var onDismissDistrict: ((Bool) -> Void)?
+	private var onEditButtonTap: (() -> Void)?
+	private var onAccessibilityFocus: (() -> Void)?
+	private var viewModel: AddStatisticsCardsViewModel?
+	private var model: AddLocalStatisticsModel?
+}
+
+enum CreatedLocalStatisticsState {
+	case empty // 0
+	case NotYetFull // 0 < number of local statistics cards < 5
```

Comment:
notYetFull

Reply:
I would ditch the whole state which can lead to complications if not updated correctly and create a computed var:
```swift
var canAddMore: Boo { return currentCount < threshold }
```
States are bad!

Reply:
@whiskey we have 3 have three states o:
1- show only add button if there number of cards = 0
2- show both add and edit buttons when 0 < number < 5
3- disable add button but leave the edit button enabled when 5 < number

Reply:
I know the use case but don't think it's the best solution here because if I miss out to update the state after removing a cell, it might get corrupted. Let's leave it for now but I'd like to test out my 'computed approach' as well and see if this is easy to implement w/o any dependencies to the `store`.